inventory/main/src/test/java/org/opennms/horizon/inventory/component/TaskSetResultsConsumerTest.java

```
@@ -0,0 +1,224 @@
+
+        var testTenantLocationSpecificTaskSetResults =
+            TenantLocationSpecificTaskSetResults.newBuilder()
+                .setTenantId("x-tenant-id-x")
+                .setLocationId(TEST_LOCATION_ID_TEXT)
+                .addResults(testResult1)
+                .addResults(testResult2)
+                .addResults(testResult3)
+                .build();
+
+        //
+        // Execute
+        //
+        var messageBytes = testTenantLocationSpecificTaskSetResults.toByteArray();
+        target.receiveMessage(messageBytes);
+
+        //
+        // Verify the Results
+        //
+        Mockito.verify(mockScannerResponseService).accept("x-tenant-id-x", TEST_LOCATION_ID, testScannerResponse1);
+        Mockito.verify(mockScannerResponseService).accept("x-tenant-id-x", TEST_LOCATION_ID, testScannerResponse2);
+        Mockito.verify(mockScannerResponseService).accept("x-tenant-id-x", TEST_LOCATION_ID, testScannerResponse3);
+        Mockito.verifyNoMoreInteractions(mockScannerResponseService);
+    }
+
+    @Test
+    void testEmptyTenantId() {
+        //
+        // Setup Test Data and Interactions
+        //
+        var testScannerResponse =
+            ScannerResponse.newBuilder()
+                .build();
+
+        var testResult = makeSacnnerTaskResult("x-task-id-001-x", testScannerResponse);
+
+        var testTenantLocationSpecificTaskSetResults =
+            TenantLocationSpecificTaskSetResults.newBuilder()
+                .setTenantId("")
+                .setLocationId(TEST_LOCATION_ID_TEXT)
+                .addResults(
+                    testResult
+                )
+                .build();
+
+        //
+        // Execute
+        //
+        try (LogCaptor logCaptor = LogCaptor.forClass(TaskSetResultsConsumer.class)) {
+            var messageBytes = testTenantLocationSpecificTaskSetResults.toByteArray();
+            target.receiveMessage(messageBytes);
```

You can assert exception being thrown instead of capturing a log message.